addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1148,6 +1149,349 @@ registry.anchorSlide = publicWidget.Widget.extend({
     },
 });
 
+
+registry.countdown = publicWidget.Widget.extend({
+    selector: '.s_countdown',
+    xmlDependencies: ['/website/static/src/xml/website.s_countdown.xml'],
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    start: function () {
+        this.hereBeforeTimerEnds = false;
+        this.endAction = this.$el[0].dataset.endAction;
+        this.endTime = parseInt(this.$el[0].dataset.endTime);
+        this.size = this.$el[0].dataset.size;
+        this.display = this.$el[0].dataset.display;
+
+        this.layout = this.$el[0].dataset.layout;
+        this.shape = this.$el[0].dataset.shape;
+        this.shapeBackground = this.$el[0].dataset.shapeBackground;
+        this.progressBarStyle = this.$el[0].dataset.progressBarStyle;
+        this.progressBarWeight = this.$el[0].dataset.progressBarWeight;
+
+        this.diff = {
+            'days': {
+                'canvas': this.$('.s_countdown_days')[0],
+                'total': 15, // TODO RATIO DAYS ? Based on how many days ?
+                'visible': this._isUnitVisible('d'),
+                'label': _t('Days'),
+            },
+            'hours': {
+                'canvas': this.$('.s_countdown_hours')[0],
+                'total': 24,
+                'visible': this._isUnitVisible('h'),
+                'label': _t('Hours'),
+            },
+            'minutes': {
+                'canvas': this.$('.s_countdown_minutes')[0],
+                'total': 60,
+                'visible': this._isUnitVisible('m'),
+                'label': _t('Minutes'),
+            },
+            'seconds': {
+                'canvas': this.$('.s_countdown_seconds')[0],
+                'total': 60,
+                'visible': this._isUnitVisible('s'),
+                'label': _t('Seconds'),
+            },
+        };
```

Comment:
I don't like relying on the fact JS objects are sorted but in fact: why using an object at all? ;) Why adding a config to say if it is visible for each unit then on each rendering checking that `visible` attribute? Why not something like this:
```javascript
this.diff = [];
if (this.isUnitVisible('d')) {
    this.diff.push({
        canvas: this.$('.s_countdown_days')[0],
        total: 15,
        label: _t("Days"),
    });
}
if (...) { ... }
if (...) { ... }
if (...) { ... }
```
That way you only have what you need to iterate over in the `_render` method. Also, why are the canvas already existing? Should not they be created here too ?

Reply:
- I used an object to identify which unit is which, but by adding `nbSeconds` in every array entry, we don't need to know which entry is which and we can make the code generic in the `_render()` method indeed :+1: 
- About the canvas, I don't have any preferences about having 4 empty canvas in the snippet template or having an empty snippet and adding the canvas in the start method.
I guess you prefer the second option.
Change done, `canvas: $('<canvas/>').appendTo(this.$('.s_countdown_canvas_wrapper'))[0],`